org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java

```diff
@@ -521,26 +521,20 @@ private InputStream fetchFromUrlSpecific(String source, boolean optional) throws
 
   private InputStreamWithSrc loadFromCIBuild(String id) throws IOException {
     checkBuildLoaded();
-    if (ciList.containsKey(id)) {
-      InputStream stream = fetchFromUrlSpecific(Utilities.pathURL(ciList.get(id), "package.tgz"), false);
-      return new InputStreamWithSrc(stream, Utilities.pathURL(ciList.get(id), "package.tgz"), "current");
+    if (packageRepos.containsKey(id)) {
+      InputStream stream = fetchFromUrlSpecific(Utilities.pathURL((String)packageRepos.get(id), "package.tgz"), false);
+      return new InputStreamWithSrc(stream, Utilities.pathURL((String)packageRepos.get(id), "package.tgz"), "current");
     } else if (id.startsWith("hl7.fhir.r5")) {
       InputStream stream = fetchFromUrlSpecific(Utilities.pathURL("http://build.fhir.org", id + ".tgz"), false);
       return new InputStreamWithSrc(stream, Utilities.pathURL("http://build.fhir.org", id + ".tgz"), "current");
     } else {
-      throw new FHIRException("The package '" + id + "' has no entry on the current build server");
+      throw new FHIRException("The package '" + id + "' has no entry on https://github.com/hl7/package-info and an entry is required to use #current.  Please submit a pull request against the package-info.json file.");
     }
   }
 
   private String getPackageUrlFromBuildList(String packageId) throws IOException {
     checkBuildLoaded();
-    for (JsonElement n : buildInfo) {
-      JsonObject o = (JsonObject) n;
-      if (packageId.equals(JSONUtil.str(o, "package-id"))) {
-        return JSONUtil.str(o, "url");
-      }
-    }
-    return null;
+    return (String)packageRepos.get(packageId);
```

Comment:
Again, does this actually require the cast?

Reply:
Fixed